src/components/withNavigationFallback.js

```diff
@@ -1,47 +1,43 @@
-import React, {Component} from 'react';
+import React, {forwardRef, useContext} from 'react';
 import {NavigationContext} from '@react-navigation/core';
 import getComponentDisplayName from '../libs/getComponentDisplayName';
 import refPropTypes from './refPropTypes';
 
 export default function (WrappedComponent) {
-    class WithNavigationFallback extends Component {
-        render() {
-            if (!this.context) {
-                return (
-                    <NavigationContext.Provider
-                        value={{
-                            isFocused: () => true,
-                            addListener: () => () => {},
-                            removeListener: () => () => {},
-                        }}
-                    >
-                        <WrappedComponent
-                            // eslint-disable-next-line react/jsx-props-no-spreading
-                            {...this.props}
-                            ref={this.props.forwardedRef}
-                        />
-                    </NavigationContext.Provider>
-                );
-            }
+    function WithNavigationFallback(props) {
+        const context = useContext(NavigationContext);
 
-            return (
+        return !context ? (
+            <NavigationContext.Provider
+                value={{
+                    isFocused: () => true,
+                    addListener: () => () => {},
+                    removeListener: () => () => {},
+                }}
+            >
```

Comment:
being a provider, it is common to wrap the objects from the value into a `useMemo` if not objects create a new reference every rerender and that will produce a chain of rerenders to all the children